layered-architectures-module-2/Carting.DAL/Repositories/Implementations/CartRepository.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Carting.DAL.Models;
+using Carting.DAL.Repositories.Interfaces;
+using MongoDB.Driver;
+
+namespace Carting.DAL.Repositories.Implementations;
+
+public class CartRepository : MongoRepositoryBase<Cart>, ICartRepository
+{
+    public CartRepository(IMongoDatabase mongoDatabase) : base(mongoDatabase)
+    {
+    }
+
+    public virtual Task<Cart> GetCartByExternalIdAsync(int id)
+    {
+        return _itemCollection.FindAsync(Builders<Cart>.Filter.Eq(c => c.ExternalId, id)).Result.SingleOrDefaultAsync();
+    }
+
+    public virtual Task DeleteByExternalIdAsync(int id)
+    {
+        return _itemCollection.DeleteOneAsync(Builders<Cart>.Filter.Eq(c => c.ExternalId, id));
+    }
+
+    public async Task AddCartItemAsync(int cartExternalId, CartItem cartItem)
+    {
+        var cart = await _itemCollection.FindAsync(cart => cart.ExternalId == cartExternalId).Result
+            .SingleOrDefaultAsync();
```

Comment:
I think it's not mandatory to use `SingleOrDefaultAsync` here. just `SingleOrDefault` instead. I'd expect that `FindAsync` would make the call async.
Also use `Find` method instead if you make the call synchronous, don't use Result over Async version

Reply:
Fixed